app/controllers/topics_controller.rb

```diff
@@ -288,6 +288,18 @@ def update
       else
         return render_json_error(I18n.t('category.errors.not_found'))
       end
+
+      if category && topic_tags = params[:tags] || topic.tags.pluck(:name)
+        allowed_tags = category.tags.pluck(:name)
+
+        if !(topic_tags.blank? || allowed_tags.blank?)
```

Comment:
How about using the opposite condition?

```suggestion
        if topic_tags.present? && allowed_tags.present?
```